### PR TITLE
Make sure ilastik errors when trying to access missing data in headless

### PR DESCRIPTION
**Pull request by k-dominik**
When running ilastik in headless, we do not require training data to be present. However, ilastik will try to train its classifier, if it has not been saved to the project file. This can lead to training from all zeros when data is inaccessible.
This is of course not acceptable.

fixes https://github.com/ilastik/ilastik/issues/2299

### CODE REVIEW

File: lazyflow/operators/valueProviders.py [hunk 1]
```diff
@@ -522,8 +529,10 @@ def setupOutputs(self):
         pass
 
     def execute(self, slot, subindex, roi, result):
-        result[...] = 0
-        return result
+        raise MissingDataAccessError(
```
Comment (emilmelnikov) [inline lazyflow/operators/valueProviders.py:532]: The error message could be moved into `MissingDataAccessError.__init__()`.

Also, the operator name is slightly more general than the exception message. Maybe rename it to something like `OpMissingDataSource`?

**Review by emilmelnikov — APPROVED** [review]
The error message could be moved into `MissingDataAccessError.__init__()`.

Also, the operator name is slightly more general than the exception message. Maybe rename it to something like `OpMissingDataSource`?

**Comment by emilmelnikov** [comment]
Alternatively, we could generalize the op by lifting the error message to the op constructor.

**Comment by Tomaz-Vieira** [comment]
I feel rather unconfortable having a DataSource that just throws an exception... It feels like us admitting defeat.

Maybe we can do this kind of stuff right at the start of the workflow, as it is parsing the command line parameters and determining whether it is going to train or not?

**Comment by k-dominik** [comment]
> I feel rather unconfortable having a DataSource that just throws an exception... It feels like us admitting defeat.
> 
> Maybe we can do this kind of stuff right at the start of the workflow, as it is parsing the command line parameters and determining whether it is going to train or not?

I agree that (in the current framework) we might want to add, maybe a method to the exporting operator that is overridden by every workflow and that can do certain checks and produce a meaningful message. Nevertheless - if all goes wrong there should still be a point of failure, so in the current architecture we need this.

**Review by emilmelnikov — APPROVED** [review]

**Pull request merged**